### DISCUSSION TITLE
Are the Fluent themes included in .NET 9 in a stable state?

### DISCUSSION THREAD

**Discussion by aquinn39** [dotnet/wpf#10072] [Q&A]
According to the documentation, the ThemeMode property is still in evaluation. The documentation seems to imply that opting into the Fluent styles by specifying them in Application.xaml is supported, however.

However there are a bunch of issues still with the Fluent themes, including that the high contrast one is completely broken and crashes the app when applied. Are these themes supposed to be in a stable state? Are they expected to be able to be used in production/release software?

**Comment by singhashish-wpf** [comment]
@aquinn39 the high-contrast issue is being worked upon as of now. We are trying our best to fix other reported issues and ship them with upcoming servicing releases.

**Reply by aquinn39** [reply]
Ok that's good to hear, but my main question is whether or not the new themes are considered safe to use in production, or if they are still considered in evaluation/experimental status.

**Reply by Balkoth** [reply]
I would not use this in a production app, the red warning boxes in the documentation make it pretty clear that this is a work in progress. Things can change any time and when they do you have to rewrite or search for alternatives, or work around bugs you have no control over.

**Reply by aquinn39** [reply]
I see. Also where are these red boxes in the docs you're talking about?

**Reply by Balkoth** [reply]
[See here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/whats-new/net90?view=netdesktop-9.0#thememode)

**Reply by aquinn39** [reply]
Ahh right. See that's what my question is about - if you look [here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/whats-new/net90?view=netdesktop-9.0#apply-the-theme), it specifies how to set the theme by specifying the resources in the app's resources. In this section, there is nothing saying that doing it that way is in evaluation state. There is then a new header 'ThemeMode' which specifies how to set it using the ThemeMode property and under this heading it say that the ThemeMode property API is in evaluation state. Additionally, if you use ThemeMode in code, you get a warning saying that it is in evaluation state, but not if you specify the themes in the app's resources.

So this gave me the impression that only using the ThemeMode property was in evaluation state (maybe because they hadn't finalised the name of it or the enum values and/or maybe the code behind this API in terms of how it adds the themes to MergedDictionaries was also not finalised, hence why they suggest doing it manually).

**Reply by miloush** [reply]
Yeah it looks very much like https://github.com/dotnet/wpf/pull/10051, I didn't realize that text had already been published before and as I noted in the PR it certainly understates how not-final this is. (You can also judge that from all the issues coming up. It simply is an addition that couldn't be done in one release cycle.)
